homeassistant/components/qnap_qsw/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""The QNAP QSW integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioqsw.const import QSD_FIRMWARE, QSD_FIRMWARE_INFO, QSD_PRODUCT, QSD_SYSTEM_BOARD
+from aioqsw.localapi import ConnectionOptions, QnapQswApi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import QswUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class QswEntity(CoordinatorEntity[QswUpdateCoordinator]):
+    """Define an QNAP QSW entity."""
+
+    def __init__(
+        self,
+        coordinator: QswUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self._attr_device_info: DeviceInfo = {
```

Comment:
```suggestion
        self._attr_device_info = DeviceInfo(
```
...

from homeassistant.helpers.entity import DeviceInfo